### DISCUSSION TITLE
Search/filter LineEdit consistency.

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#7074] [Editor]
Search/filter type LineEdit controls in the Editor mostly use a "clear" button and a "search" (magnify) icon but not all.  The "find bars" in the ScriptEditor and EditorHelp don't.  For the sake of consistency, why don't they all employ the same interface in this regard ?

**Comment by theraot** [comment]
**I agree** on adding the clear button and the magnify glass icon to the search bars.

As per why they don't use the same interface, is because filter and search are not the same thing. Do the search in the script editor hide lines of the script? No it does not. So the prior and next occurrence of what you are searching for aren't the prior and next line, thus it makes sense to have buttons to navigate to them.

Now, the counterpart of this and that https://github.com/godotengine/godot-proposals/discussions/7076 - Why not make ESC clear filters?

**Comment by ghost** [comment]
I'm really only talking about adding the "clear" button and the right "icon" to the FindReplaceBar LineEdit in the ScriptEditor and the search LineEdit in the EditorHelp DOC/class reference.  I've written a plug-in that does this and it makes everything look consistent.  As for my other idea of using CTRL + F to open close these search LineEdits, I'm now no longer sure after implementing this.  It seems to affect the flow of things, but maybe that's just because I'm used to closing the search-bars via the ESC key.

  Anyway, my plug-in has also employed the use of the ESC key to clear the LineEdits because they are primarily a keyboard control, therefore, I thought that surely the "clear" button should be capable of being "pressed" via the ESC key.  It actually works pretty well.

  The "clear" button and the right "icon" are used on "filter" type controls, just look at the Scripts panel for filtering Scripts and the filtering Methods panel, aswell as the filter messages LineEdit in the "Output" panel, so in fact there a quite a lot that use the same interface.
